src/main/java/com/iexec/worker/compute/ComputeManagerService.java

```diff
@@ -74,8 +79,46 @@ public boolean downloadApp(TaskDescription taskDescription) {
         if (!isDockerType || taskDescription.getAppUri() == null) {
             return false;
         }
+
+        final long pullTimeout = computeImagePullTimeout(taskDescription);
         return dockerService.getClient(taskDescription.getAppUri())
-                .pullImage(taskDescription.getAppUri());
+                .pullImage(taskDescription.getAppUri(), Duration.of(pullTimeout, ChronoUnit.MINUTES));
+    }
+
+    /**
+     * Computes image pull timeout depending on task max time execution.
+     * This should depend on task category (XS, S, M, L, XL).
+     * <p>
+     * Current formula is: {@literal 10 * log(maxExecutionTime * 10)},
+     * with {@literal maxExecutionTime} being expressed in minutes.
+     * The result is rounded to the nearest integer.
+     * <p>
+     * Examples of timeout depending on category
+     * - with default values of 5 for min and 30 for max:
+     * <ul>
+     *     <li>XXS: 10 * log(    20 / 10) =  3 minutes => 5 minutes because of min</li>
+     *     <li>XS : 10 * log(    50 / 10) =  7 minutes</li>
+     *     <li>S  : 10 * log(   200 / 10) = 13 minutes</li>
+     *     <li>M  : 10 * log(   600 / 10) = 18 minutes</li>
+     *     <li>L  : 10 * log(  1800 / 10) = 23 minutes</li>
+     *     <li>XL : 10 * log(  6000 / 10) = 28 minutes</li>
+     *     <li>XXL: 10 * log(100000 / 10) = 40 minutes => 30 minutes because of max</li>
+     * </ul>
+     * If computed timeout duration is lower than {@link DockerRegistryConfiguration#getMinPullTimeout()},
+     * then final timeout duration is {@link DockerRegistryConfiguration#getMinPullTimeout()}.
+     * <br>
+     * If computed timeout duration is greater than {@link DockerRegistryConfiguration#getMaxPullTimeout()},
+     * then final timeout duration is {@link DockerRegistryConfiguration#getMaxPullTimeout()}.
+     */
+    long computeImagePullTimeout(TaskDescription taskDescription) {
+        final long maxExecutionTime = taskDescription.getMaxExecutionTime() / 60;
+        return Math.max(
```

Comment:
Wouldn't it be better to perform calculation once and to cache it ?
We have 5 computations that could be performed when the worker starts and be stored in a map.
We would just need to fetch the correct value at runtime.

Reply:
Right, see https://github.com/iExecBlockchainComputing/iexec-worker/pull/442/commits/0aed6645743bc1ab69aa02c20d5716da3e07959e.

Reply:
`new HashMap<>(5)` simply creates an HashMap whose default capacity is of 5. It's automatically extended if required. So I think it's better to keep it at 5 and we'll update it if it is really needed. That'll save some (very little) memory. Do you agree?

Reply:
You are right, but default size is 16, pretty minor performance update..!
We can keep it if you like it ;)

Reply:
Thanks or the caching with a `Map`.
I think even the division by 60 could be avoided.
There's not much difference in the storage size and it avoids a repetitive computation.

Reply:
I have thought about the division. But as pointed out by @jeremyjams [in this comment](https://github.com/iExecBlockchainComputing/iexec-worker/pull/442#discussion_r834461584), we should strive to have a single unit. That would be confusing to have the keys as seconds while the values are minutes.